src/js/structs/Tree.js

```diff
@@ -127,11 +127,17 @@ module.exports = class Tree extends List {
         }, {ids: filter.ids});
       }
 
-      if (filter.name) {
-        let tree =
-          new this.constructor(Object.assign({},
-            this, {items: services, filterProperties: this.getFilterProperties()}));
-        services = tree.filterItemsByText(filter.name).getItems();
+      if (filter.id) {
+        let filterProperties = Object.assign({}, this.getFilterProperties(), {
+          name: function (item) {
+            return item.id;
```

Comment:
All service instance (frameworks as well as applications) do provide an id (`getId`) and name (`getName`) getter – let's use them.